crates/mcp/src/surfpool/set_token_account.rs

```diff
@@ -0,0 +1,317 @@
+/// * `SetTokenAccountResponse`: Contains either details of the successful account update (including new wallet details if generated) or an error message.                             
+pub fn run(
+    surfnet_address: String,
+    wallet_address_opt: Option<String>,
+    token_identifier_str: String, // Can be "SOL", a mint address, or a known symbol
+    token_amount_opt: Option<u64>,
+) -> SetTokenAccountResponse {
+    let client = Client::new();
+    let rpc_url = surfnet_address;
+
+    // Determine the owner's public key. If no wallet address is provided, generate a new keypair.
+    // The keypair itself is stored if generated, so the secret key can be returned.
+    let owner_keypair: Option<Keypair> = if wallet_address_opt.is_none() {
+        Some(Keypair::new())
+    } else {
+        None
+    };
+    let owner_pubkey_str = wallet_address_opt
+        .clone()
+        .unwrap_or_else(|| owner_keypair.as_ref().unwrap().pubkey().to_string());
+
+    // Validate the owner public key string.
+    let owner_pubkey = match Pubkey::try_from(owner_pubkey_str.as_str()) {
+        Ok(pk) => pk,
+        Err(_) => {
+            return SetTokenAccountResponse::error(format!(
+                "Invalid owner wallet address: {}",
+                owner_pubkey_str
+            ))
+        }
+    };
+
+    let amount_to_set = token_amount_opt.unwrap_or(DEFAULT_TOKEN_AMOUNT);
+
+    // Determine if it's SOL, a direct mint address, or a symbol to look up.
+    let is_sol = token_identifier_str.to_uppercase() == SOL_SYMBOL;
+    let mut actual_mint_address_str: String;
+    let mut actual_token_symbol_opt: Option<String> = None;
+
+    if is_sol {
+        actual_mint_address_str = SOL_MINT_PLACEHOLDER.to_string();
+        actual_token_symbol_opt = Some(SOL_SYMBOL.to_string());
+    } else {
+        // Not SOL, try parsing as a mint address directly.
+        match Pubkey::try_from(token_identifier_str.as_str()) {
+            Ok(_mint_pubkey) => {
+                actual_mint_address_str = token_identifier_str.clone();
+                actual_token_symbol_opt = VERIFIED_TOKENS_BY_SYMBOL
+                    .iter()
+                    .find(|(_, info)| info.address == actual_mint_address_str)
+                    .map(|(sym, _)| sym.clone());
+            }
+            Err(_) => {
+                // if not a direct mint address, try looking up as a symbol in VERIFIED_TOKENS_BY_SYMBOL (case-insensitive).
+                match VERIFIED_TOKENS_BY_SYMBOL.get(&token_identifier_str.to_uppercase()) {
+                    Some(token_info) => {
+                        actual_mint_address_str = token_info.address.clone();
+                        actual_token_symbol_opt = Some(token_info.symbol.clone());
+                    }
+                    None => {
+                        return SetTokenAccountResponse::error(format!(
+                            "Invalid token identifier: '{}'. Not SOL, a valid mint address, or a known symbol.",
```

Comment:
I think this error should be something along the lines:
The token symbol provided '<symbol>' is not a verified token. Please provide the token address instead.